config/settings.rb

```diff
@@ -29,6 +29,9 @@
   SETTINGS[:fqdn] ||= Facter.value(:fqdn)
 end
 
+SETTINGS[:hosts] = [] unless SETTINGS[:hosts].is_a?(Array)
+SETTINGS[:hosts].map! { |h| h == 'FQDN' ? SETTINGS[:fqdn] : h }
```

Comment:
It's guaranteed by this point that `SETTINGS[:fqdn]` has a value (because of the `unless` block above.)